src/MissionEditor/MissionEditor.qml

```diff
@@ -484,8 +501,7 @@ QGCView {
                             onClicked:  setCurrentItem(object.sequenceNumber)
 
                             onRemove: {
-                                itemDragger.clearItem()
-                                controller.removeMissionItem(object.sequenceNumber)
```

Comment:
The issues with Survey delete is that there is an out of range error when I delete one of some surveys which have been edited out of order. I'm sure that it is crashing due to incorrect index.